src/controls/controls.render.ts

```diff
@@ -69,18 +91,20 @@ export function renderCircleControl(
  * @param {CanvasRenderingContext2D} ctx context to render on
  * @param {Number} left x coordinate where the control center should be
  * @param {Number} top y coordinate where the control center should be
- * @param {Object} styleOverride override for fabric.Object controls style
- * @param {fabric.Object} fabricObject the fabric object for which we are rendering controls
+ * @param {Object} styleOverride override for FabricObject controls style
+ * @param {FabricObject} fabricObject the fabric object for which we are rendering controls
  */
 export function renderSquareControl(
-  ctx,
-  left,
-  top,
-  styleOverride,
-  fabricObject
+  this: Control,
```

Comment:
i think this `this` isn't needed. is it? we are calling this with `call`

Reply:
It is for TS to cast the this context of the method
Doesn't affect anything else AFAIK
I think these methods belong in the control class

Reply:
so is not a real argument? i can call the function like `renderSquareControl(ctx, left... )` ?

Reply:
I see when built it gets removed.
Weird that being a type annotation looks like a real argument.
How do someone know just reading at the code?

Reply:
Weird indeed